File: app/src/main/java/com/mickstarify/fortunemod/Quote.java


package com.mickstarify.fortunemod;

/**
 * Copyright Michael Johnston
 * Created by michael on 4/12/14.
 */
public class Quote {
    public int id;
    public String quote;
    public String category;
    public boolean isOffensive;

    public Quote (int id, String quote, String category, boolean isOffensive){
        this.id = id;
        this.quote = quote;
        this.category = category;
        this.isOffensive = isOffensive;
    }

}

File: app/src/main/kotlin/com/mickstarify/fortunemod/Quote.kt
package com.mickstarify.fortunemod

import android.os.Parcel
import android.os.Parcelable

/**
 * Copyright Michael Johnston
 * Created by michael on 4/12/14.
 */
data class Quote(var id: Int,
                 var quote: String,
                 var category: String,
                 var isOffensive: Boolean) : Parcelable {

    constructor(parcel: Parcel) : this(
            parcel.readInt(),
            parcel.readString(),
            parcel.readString(),
            if (parcel.readInt() == 1) {
                true
            } else {
                false
            }
    )

    override fun describeContents(): Int = 0
    override fun writeToParcel(dest: Parcel,
                               flags: Int) {

        dest.writeInt (id)
        dest.writeString(quote)
        dest.writeString(category)
        dest.writeInt(if (isOffensive) {
            1
        } else {
            0
        })
    }

    companion object {
        val CREATOR = object : Parcelable.Creator<Quote> {
            override fun createFromParcel(parcel: Parcel): Quote {
                return Quote (parcel)
            }

            override fun newArray(size: Int): Array<Quote?> {
                return arrayOfNulls(size)
            }

        }
    }
}
